runtime/zeitgeist/src/integration_tests/xcm/tests/transfers.rs

```diff
@@ -0,0 +1,336 @@
+        assert_eq!(
+            Tokens::free_balance(FOREIGN_ZTG_ID, &BOB.into()),
+            bob_initial_balance - transfer_amount
+        );
+    });
+
+    Zeitgeist::execute_with(|| {
+        // Verify that ALICE now has initial balance + amount transferred - fee
+        assert_eq!(
+            Balances::free_balance(&ALICE.into()),
+            alice_initial_balance + transfer_amount - ztg_fee(),
+        );
+
+        // Verify that the reserve has been adjusted properly
+        assert_eq!(
+            Balances::free_balance(&sibling_account()),
+            sibling_sovereign_initial_balance - transfer_amount
+        );
+
+        // Verify that fees (of native currency) have been put into treasury
+        assert_eq!(
+            Balances::free_balance(ZeitgeistTreasuryAccount::get()),
+            treasury_initial_balance + ztg_fee()
+        )
+    });
+}
+
+#[test]
+fn transfer_ksm_from_relay_chain() {
+    let transfer_amount: Balance = ksm(1);
+
+    Zeitgeist::execute_with(|| {
+        register_foreign_parent(None);
+    });
+
+    KusamaNet::execute_with(|| {
+        assert_ok!(kusama_runtime::XcmPallet::reserve_transfer_assets(
+            kusama_runtime::Origin::signed(ALICE.into()),
+            Box::new(Parachain(zeitgeist::ID).into().into()),
+            Box::new(Junction::AccountId32 { network: NetworkId::Any, id: BOB }.into().into()),
+            Box::new((Here, transfer_amount).into()),
+            0
+        ));
+    });
+
+    Zeitgeist::execute_with(|| {
+        assert_eq!(
+            Tokens::free_balance(FOREIGN_PARENT_ID, &BOB.into()),
+            transfer_amount - ksm_fee()
+        );
+    });
+}
+
+#[test]
+fn transfer_ksm_to_relay_chain() {
+    let transfer_amount: Balance = ksm(1);
+    transfer_ksm_from_relay_chain();
+
+    Zeitgeist::execute_with(|| {
+        let initial_balance = Tokens::free_balance(FOREIGN_PARENT_ID, &ALICE.into());
+        assert!(initial_balance >= transfer_amount);
+
+        assert_ok!(XTokens::transfer(
+            Origin::signed(ALICE.into()),
+            FOREIGN_PARENT_ID,
+            transfer_amount,
+            Box::new(
+                MultiLocation::new(
+                    1,
+                    X1(Junction::AccountId32 { id: BOB, network: NetworkId::Any })
+                )
+                .into()
+            ),
+            4_000_000_000
+        ));
+
+        assert_eq!(
+            Tokens::free_balance(FOREIGN_PARENT_ID, &ALICE.into()),
+            initial_balance - transfer_amount
+        )
+    });
+
+    KusamaNet::execute_with(|| {
+        assert_eq!(kusama_runtime::Balances::free_balance(&BOB.into()), 999834059328);
+    });
+}
+
+#[test]
+fn transfer_ztg_to_sibling_with_custom_fee() {
+    TestNet::reset();
+
+    let alice_initial_balance = ztg(10);
+    // 10x fee factor, so ZTG has 10x the worth of sibling currency.
+    let fee_factor = 100_000_000_000;
```

Comment:
The fee factor in the asset registry is denominated using the fixed point decimal representation of the currency in question. Since ZTG uses `10_000_000_000` as a base, 10x fee factor is denominated as `100_000_000_000`. For Kusama it would be `10_000_000_000_000`, since it uses 12 fractional decimal places.
I went from one base to another plenty of times trying to figure out whether we should use a fixed number of decimal fractional places here that we use natively or if we should denote it based on the base of the currency in question. I think both approaches have their cons an pros. Using the same number of fractional places in all cases reduces the error-proneness when adding new tokens (that's what I like to think at least). However, being able to denote the fee using the base that the parachain in question uses allows for greater portability of (parts of) the asset registry because it can be interpreted out of context (you don't have to know the base of the executing chain).